crates/stellar-scaffold-macro/src/asset.rs

```diff
@@ -97,75 +141,70 @@ mod test {
-                | (Futurenet, "CAIGIAHVHIPSS2OAFGYBPKUG2DFU5AIFE5FZM24UBVCWFJIQZNYXKPE7")
-                | (Mainnet, "CCW67TSZV3SSS2HXMBQ5JFGCKJNXKZM7UQUWUZPUTHXSTZLEO7SJMI75") => {}
-                (x, s) => panic!("Unexpected network {x:?} with asset {s}"),
-            }
-        }
+        assert_eq!(get_serialized_asset(USDC).unwrap().0, "USDC");
+        assert_eq!(
+            get_serialized_asset(USDC).unwrap().1,
+            [
+                0, 0, 0, 1, 85, 83, 68, 67, 0, 0, 0, 0, 59, 153, 17, 56, 14, 254, 152, 139, 160,
+                168, 144, 14, 177, 207, 228, 79, 54, 111, 125, 190, 148, 107, 237, 7, 114, 64, 247,
+                246, 36, 223, 21, 197
+            ]
+        );
     }
 
     #[test]
     fn native_client() {
         let lit: syn::LitStr = syn::parse_quote!("native");
         let expected = quote! {
-            #[allow (non_upper_case_globals)]
-            pub(crate) mod native {
+            #[allow(non_upper_case_globals)]
+            pub(crate) mod test_native {
                 use super::*;
-                /// Contract id for the Stellar Asset Contract
-                pub fn contract_id(env: &soroban_sdk::Env) -> soroban_sdk::Address {
-                    soroban_sdk::Address::from_str(&env, "CDLZFC3SYJYDZT7K67VZ75HPJVIEUVNIXF47ZG2FB2RMQQVU2HHGCYSC")
-                }
+
                 /// Create a Stellar Asset Client for the asset which provides an admin interface
-                pub fn stellar_asset_client<'a>(env: &soroban_sdk::Env) -> soroban_sdk::token::StellarAssetClient<'a> {
-                    soroban_sdk::token::StellarAssetClient::new(&env, &contract_id(env))
+                pub fn stellar_asset_client<'a>(env: &soroban_sdk::Env, sac:&soroban_sdk::testutils::StellarAssetContract) -> soroban_sdk::token::StellarAssetClient<'a> {
+                    soroban_sdk::token::StellarAssetClient::new(&env, &sac.address())
                 }
                 /// Create a Stellar Asset Client for the asset which provides an admin interface
-                pub fn token_client<'a>(env: &soroban_sdk::Env) -> soroban_sdk::token::TokenClient<'a> {
-                    soroban_sdk::token::TokenClient::new(&env, &contract_id(env))
+                pub fn token_client<'a>(env: &soroban_sdk::Env, sac:&soroban_sdk::testutils::StellarAssetContract) -> soroban_sdk::token::TokenClient<'a> {
+                    soroban_sdk::token::TokenClient::new(&env, &sac.address())
+                }
+
+                /// Registers a new SAC contract (to use in unit tests only)
+                pub fn register(env: &soroban_sdk::Env, admin: &soroban_sdk::Address) -> soroban_sdk::testutils::StellarAssetContract {
+                    let sac = env.register_stellar_asset_contract_v2(admin.clone());
+                    let cl = stellar_asset_client(env, &sac);
+                    env.mock_all_auths();
+                    cl.mint(admin, &1_000_000_000_i128);
+                    sac
```

Comment:
We should also reset the auths by passing an empty array to set auths. This way the caller won't have this side effect.